lib/kubeclient/informer.rb

```diff
@@ -65,17 +65,17 @@ def fill_cache
     end
 
     def watch_to_update_cache
-      watcher = @client.watch_entities(@resource_name, watch: true, resource_version: @started)
+      @watcher = @client.watch_entities(@resource_name, watch: true, resource_version: @started)
       stop_reason = 'disconnect'
 
       # stop watcher without using timeout
       Thread.new do
         sleep(@reconcile_timeout)
         stop_reason = 'reconcile'
-        watcher.finish
+        @watcher.finish
```

Comment:
Consider adding `@watcher = nil` to release the watcher instance to GC after this method exits? (Possibly in `ensure` since this runs in worker thread that can be `.kill`ed)

Not sure if this matters for anything.  A stopped WatchStream instance is not using much resources, and at most 1 could remain on an inactive informer.

What might matter more is another scenario, unrelated to this PR: if informer worker thread is killed, this `@watcher.finish` might not run, possibly leaking an open watch socket?  Does that need an `ensure` somewhere?  Or maybe cleaner to add `@watcher&.finish` in `#stop_worker` (possible now that you made it an instance variable)

Reply:
This later scenario might be exactly what happened in this CI run, .finish wasn't called:
https://github.com/ManageIQ/kubeclient/actions/runs/3560257866/jobs/5980172236#step:6:223

Reply:
Looking at `WatchStream` implementation, I don't think releasing `@watcher` instance is important.  It in turn holds an `HTTP::Client` instance, but .finish force-closes it, so there should be no issue.